core/src/main/java/com/datastax/oss/driver/internal/core/protocol/BuiltInCompressors.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.protocol;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.protocol.internal.Compressor;
+import io.netty.buffer.ByteBuf;
+
+public class BuiltInCompressors {
+
+  public static Compressor<ByteBuf> get(String name, DriverContext context) {
```

Comment:
The whole thing relies on the fact that this will be the unique entry point to instantiate a compressor. Maybe we could make the constructors package-private to emphasize that.

I think that's reasonable, compressors are internal anyway and clients have no reason to instantiate them directly. And if someone wants to plug a custom compressor, they'd have to extend the context anyway, so we can hope that they see the substitution stuff and adapt it appropriately.

Reply:
Agreed, I don't see this as a major concern.  As mentioned somebody would have to have a custom DriverContext to even exercise the problem here and in that case it doesn't seem super-surprising that your results might vary if you diverge from the patterns in place.

_Perhaps_ it's worth a comment here to say something to the effect of "BuiltInCompressors exists to allow for easy substitutions on GraalVM in cases where underlying dependent libraries aren't present".  Or, you know, something saying the same thing but saying it better. :feelsbadman:  I don't think it's essential but it might at least help a little bit.

I don't really love the idea of making compressor constructors package-private... that feels more disconnected from the problem we're actually trying to solve here.